File: 118.pascals-triangle.cs
// @leet start
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
      var result = new List<IList<int>>();
      for (int i = 0; i < numRows; i++) {
        var row = new List<int>();
        for (int j = 0; j <= i; j++) {
          if (j == 0 || j == i) {
            row.Add(1);
          } else if (i > 1) {
            row.Add(result[i - 1][j - 1] + result[i - 1][j]);
          }
        }
        result.Add(row);
      }
      return result;
    }
}
// @leet end

File: 118.pascals-triangle.kt
// @leet start
class Solution {
    fun generate(numRows: Int): List<List<Int>> {
      val result = mutableListOf<List<Int>>()
      for (i in 1..numRows) {
        val row = mutableListOf<Int>()
        for (j in 1..i) {
          if (j == 1 || j == i) { row.add(1) }
          else if (i > 2) {
            row.add(result[i - 2][j - 2] + result[i - 2][j - 1])
          }
        }
        result.add(row)
      }
      return result
    }
}
// @leet end
